app/build-sdk.sh

```diff
@@ -62,7 +68,17 @@ function check_last_commit {
 	return 0
 }
 
+function checkout_matching_branch {
+	app_branch=$(git_app_branch)
+	cd $SDK_DIR && $GIT checkout $app_branch 1>/dev/null 2>/dev/null
+}
+
+checkout_matching_branch
+
 SHA=$(git_sha)
+if [ "X$SHA" == "X" ]; then
```

Comment:
Wouldn't `if [ -z "$SHA" ]; then` work for this case?

Reply:
Probably... It's been years without doing "serious" bash development :stuck_out_tongue: